packages/extension-sdk/src/connect/tile/tile_sdk.spec.ts

```diff
@@ -0,0 +1,167 @@
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+import type { ExtensionHostApiImpl } from '../extension_host_api'
+import { TileSDKImpl } from './tile_sdk'
+import type { CrossFilterOptions } from './types'
+import { DashboardRunState } from './types'
+
+describe('TileSDK', () => {
+  let api: ExtensionHostApiImpl
+  beforeEach(() => {
+    api = {
+      send: jest.fn(),
+      sendAndReceive: jest.fn(),
+    } as unknown as ExtensionHostApiImpl
+  })
+
+  const makeTileSdk = () => {
+    const tileSdk = new TileSDKImpl(api)
+    tileSdk.tileHostDataChanged({
+      isEditing: true,
+      dashboardRunState: DashboardRunState.COMPLETE,
+      filters: { hello: 'world' },
+    })
+    return tileSdk
+  }
+
+  const makeEvent = (props: any = {}) => {
+    return {
+      metaKey: true,
+      pageX: 20,
+      pageY: 40,
+      type: 'click',
+      ...props,
+    }
+  }
+
+  it('constructs', () => {
+    const tileSdk = new TileSDKImpl(api)
+    const { isEditing, dashboardRunState, filters } = tileSdk.tileHostData
+    expect(isEditing).toEqual(false)
+    expect(dashboardRunState).toEqual('UNKNOWN')
+    expect(filters).toEqual({})
+  })
+
+  it('updates host data', () => {
+    const tileSdk = new TileSDKImpl(api)
+    tileSdk.tileHostDataChanged({
+      isEditing: true,
+      dashboardRunState: DashboardRunState.RUNNING,
+      filters: { hello: 'world' },
+    })
+    const { isEditing, dashboardRunState, filters } = tileSdk.tileHostData
+    expect(isEditing).toEqual(true)
+    expect(dashboardRunState).toEqual('RUNNING')
+    expect(filters).toEqual({ hello: 'world' })
+  })
+
+  it('sends add errors message ', () => {
+    const message1 = { title: 'Title1', message: 'Message1', group: 'abc' }
+    const message2 = { title: 'Title2', message: 'Message2', group: 'abc' }
+    const tileSdk = makeTileSdk()
+    tileSdk.addErrors(message1, message2)
+    expect(api.send).toBeCalledWith('TILE_ADD_ERRORS', {
+      errors: [message1, message2],
+    })
+  })
+
+  it('sends clear errors message ', () => {
+    const tileSdk = makeTileSdk()
+    tileSdk.clearErrors('abc')
+    expect(api.send).toBeCalledWith('TILE_CLEAR_ERRORS', { group: 'abc' })
+  })
+
+  it('sends tigger message ', () => {
+    const tileSdk = makeTileSdk()
+    const config = [{ hello: 'world' }]
+    const message = 'message1'
+    tileSdk.trigger(message, config, makeEvent(config))
+    expect(api.send).toBeCalledWith('TILE_TRIGGER', {
+      message,
+      config,
+      event: makeEvent(),
+    })
+  })
+
+  it('sends tigger message ', () => {
```

Comment:
Is this a duplicate test?
```suggestion
  it('sends trigger message ', () => {
```

Reply:
Yes - will delete